### PR TITLE
added nodeSelector: beta.kubernetes.io/os: linux

### PR DESCRIPTION
**Pull request by yliaog**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug
/kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
added nodeSelector to restrict the deployment to linux nodes

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
None
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by MrHohn** [comment]
@yliaog Can you update `cluster/addons/dns/kube-dns/kube-dns.yaml.in` and `cluster/addons/dns/kube-dns/kube-dns.yaml.sed` as well?

cc @prameshj

File: cluster/addons/dns-horizontal-autoscaler/dns-horizontal-autoscaler.yaml [hunk 1]
```diff
@@ -103,4 +103,6 @@ spec:
       tolerations:
       - key: "CriticalAddonsOnly"
         operator: "Exists"
+      nodeSelector:
+        beta.kubernetes.io/os: linux
```
Comment (MrHohn) [inline cluster/addons/dns-horizontal-autoscaler/dns-horizontal-autoscaler.yaml]: Should we use `kubernetes.io/os` instead? seems like the beta one is deprecated [1].

[1] https://kubernetes.io/docs/reference/kubernetes-api/labels-annotations-taints/#beta-kubernetes-io-os-deprecated

**Comment by yliaog** [comment]
the beta label is going to be removed in 1.18, we should switch before that. For now, better to use beta label as some old k8s version node does not publish non-beta label.

**Review by MrHohn — APPROVED** [review]
/lgtm
/approve

**Pull request merged**